### ISSUE TITLE
MD024 (Multiple headers with same content) should only trigger under the same nesting

### ISSUE THREAD

**Issue by deivid-rodriguez** [markdownlint/markdownlint#175]
In my opinion

```markdown
# Change log

## 1.0.0

### Bug fixes
...

### Features
...

### Bug fixes
...
```

should trigger an offense.

But

```markdown
# Change log

## 1.0.0

### Bug fixes
...

### Features
...

## 0.0.1

### Bug fixes
...
```

should not.

**Comment by tdaff** [comment]
Agree; gets very upset with API documentation too.
